ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionReceipt.java

```diff
@@ -201,6 +239,10 @@ public TransactionReceiptType getTransactionReceiptType() {
     return transactionReceiptType;
   }
 
+  public String getRevertReason() {
+    return revertReason.isPresent() ? revertReason.get() : null;
```

Comment:
can be simpler using revertReason.orElse(null)

Reply:
This function feels like a mismatch with the constructor - i.e. we can pass in an Optional Revert, but we give back a concrete ... or is this normal? (I can honestly say I don't know).

Reply:
I've changed that so the conversion from Optional<String> to String happens in the json rpc area.